### PR TITLE
Nop plugin for mattermost.

### LINKED ISSUE #12375: Create a NOP Plugin for future benchmarking
Create a plugin that implements every supported hook currently available but does nothing.

In the future, we can use this plugin as part of loadtesting to measure RPC overhead.

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-13819

### PR DESCRIPTION
**Pull request by tasdomas**
#### Summary
A nop plugin for mattermost server.

This plugin implements all defined hooks (with empty implementations).

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/12375

### CODE REVIEW

**Comment by lieut-data** [comment]
Thanks, @tasdomas! @ali-farooq0, this is the plugin we chatted about yesterday in the context of Q1 performance analysis of plugins. I'm hopeful it will help us establish a baseline on the overhead the plugin system introduces.

File: server/plugin.go [hunk 1]
```diff
@@ -1,28 +1,17 @@
...
 
-// ServeHTTP demonstrates a plugin that handles HTTP requests by greeting the world.
-func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
-	fmt.Fprint(w, "Hello, world!")
-}
+type nopPluginHooks struct{}
 
-// See https://developers.mattermost.com/extend/plugins/server/reference/
+// Check that the `plugin.Hooks` interface is implemented by `nopPluginHooks`.
+var _ plugin.Hooks = (*nopPluginHooks)(nil)
```
Comment (alifarooq0) [inline server/plugin.go:17]: ahh this is really neat!

**Comment by alifarooq0** [comment]
Excellent work @tasdomas! Looks great! especially that neat trick to check if `nopPluginHooks` satisfies the plugin hooks interface 👌.

Just curious if `NOP` abbreviation is common enough that people will know or would `NOOP` read better? (more of a question for @lieut-data i suppose :))

**Review by alifarooq0 — APPROVED** [review]

**Review by lieut-data — APPROVED** [review]
This looks great!

Re: `nop` vs. `noop`, I might lean towards the former, if only to increase the odds of pronouncing correctly. "Nupe" plugin and all that. I suppose we could do `no-op`, but I kinda like that this is also obscure given its intended audience anyway.

**Comment by lieut-data** [comment]
As this is an plugin meant to facilitate feature work and not shipping as part of the core product, I'm deferring QA review at this time. Thanks again, @tasdomas!

**Pull request merged**